Improvements to the API regarding env variables (#603)
codemagic-ci-cd/codemagic-docs: content/rest-api/applications.md

### CURL example

```bash
curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
 -H 'X-Auth-Token: <API Token>' \
 -H 'Content-Type: application/json;charset=utf-8' \
 --data '{"value": "your value"}'
```

### Response

```yaml
{
  'encrypted': 'Encrypted(Z0FBQUFBQmZMVkhwb3Q3QlJtRlVOeVFJcEJvTTRtWnZablpqMS0xN2V6dllTell1ODZSd2FUcnNqMUlZT09QY1paV0pjbVRfUlVJeDUxRWIzX1paOEZlc1dSdi1XMXlkUFVIdjNIZ2VqcE5Ja0tpMjlPWjhlSTQ9)',
}
```

## Modify application variables and secrets

> Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for applications using the `codemagic.yaml` file for configuration. They can be manually configured under **Your application > Environment variables**.

Reviewer: ```suggestion
Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings.

```

Reviewer: Perhaps add a link to the env vars doc that explains how the variables are managed in the UI so there's also info about the groups

### Fetch variables

`GET /apps/:id/variables`

> #### Curl request

Reviewer: Curl or CURL? see previous sections

```bash
curl -XGET -H 'X-Auth-Token: <API Token>' -H "Content-type: application/json" 'https://api.codemagic.io/apps/<app_id>/variables'
```

#### Response

```json
{
  [
      {
          "key": "your variable name",
          "value": "your variable value",
          "group": "your variable group",
          "secure": false
      },
  ]
}
```

### Add new variable

`POST /apps/:id/variables/`

#### Payload

> To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default the variable is encrypted. Set the value of `secure` to `false` in order to not encrypt the newly added variable. An example payload:

Reviewer: ```suggestion
To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default, the variable is encrypted. Set the value of `secure` to `false` to not encrypt the newly added variable. An example payload:

```

Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings. These variables and secrets can be accessed in your configuration file with the use of [groups](../building/environment-variable-groups).

### Fetch variables

`GET /apps/:id/variables`

Based on the application id provided, returns the configured variables.

#### Example

```bash
curl -XGET -H 'x-auth-token: <API Token>' -H "Content-type: application/json" 'https://api.codemagic.io/apps/<app_id>/variables'
```

#### Response

```json
[
  {
    "group": "production",
    "id": "619e329e0ca5fe19c3780c74",
    "key": "FOO",
    "secure": true,
    "value": "[HIDDEN]"
  }
]
```

### Add new variable

`POST /apps/:id/variables/`

#### Parameters

| **Name**        | **Type** | **Description** |
| --------------- | -------- | --------------- |
| `key` | `string` | **Required.** Name of the variable. |

> | `value` | `string` | **Required.** Value of the variable. For files base64 encode the contents. |

Reviewer 2: For binary data use `base64` to encode the contents.

| `group` | `string` | **Required.** Name of the `group` that the variable should be added to. If the group does not exist, it will be created. | 
| `secure` | `boolean` | **Optional.** By default, the variable is encrypted. Set to `false` to not encrypt the newly added variable. |

> To encode file contents and paste result to clipboard

Reviewer 2: remove it

```

## Modify application variables and secrets

Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings. These variables and secrets can be accessed in your configuration file with the use of [groups](../building/environment-variable-groups).

### Fetch variables

`GET /apps/:id/variables`

Based on the application id provided, returns the configured variables.

#### CURL request

```bash
curl -XGET -H 'x-auth-token: <API Token>' -H "Content-type: application/json" 'https://api.codemagic.io/apps/<app_id>/variables'
```

#### Response

```json
{
  [
      {
          "key": "your variable name",
          "value": "your variable value",
          "group": "your variable group",
          "secure": false,
          "id": "your variable id"
      },
  ]
}
```

### Add new variable

`POST /apps/:id/variables/`

#### Parameters

> To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data.

Reviewer 2: no need to repeat what already marked as `Required` in the table

| **Name**        | **Type** | **Description** |
| --------------- | -------- | --------------- |
| `key` | `string` | **Required.** Name of the variable. |

> | `value` | `string` | **Required.** Value of the variable. |

Reviewer 2: my prev comment about binary data and base64 got lost

| `group` | `string` | **Required.** Name of the `group` that the variable should be added to. | 
| `secure` | `boolean` | **Optional.** By default, the variable is encrypted. Set to `false` to not encrypt the newly added variable. |

> Note that if the group does not exist, it will be created.

Reviewer 2: move it to the table

```yaml
{"encrypted": "Encrypted(Z0FBQUFBQmZMVkhwb3Q3QlJtRlVOeVFJcEJvTTRtWnZablpqMS0xN2V6dllTell1ODZSd2FUcnNqMUlZT09QY1paV0pjbVRfUlVJeDUxRWIzX1paOEZlc1dSdi1XMXlkUFVIdjNIZ2VqcE5Ja0tpMjlPWjhlSTQ9)"}
```

## Modify application variables and secrets

Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings. These variables and secrets can be accessed in your configuration file with the use of [groups](../building/environment-variable-groups).

### Fetch variables

`GET /apps/:id/variables`

> #### CURL request

Reviewer 2: try to keep the same structure for all new actions. I'd recommend using:
- Parameters
- Example
- Response

```bash

> curl -XGET -H 'X-Auth-Token: <API Token>' -H "Content-type: application/json" 'https://api.codemagic.io/apps/<app_id>/variables'

Reviewer 2: `x-auth-token` in lowercase here and below

```

#### Response

```json
{
  [
      {

>           "key": "your variable name",

Reviewer 2: `id` is important but missing in all examples

          "value": "your variable value",
          "group": "your variable group",
          "secure": false
      },
  ]
}
```

### Add new variable

`POST /apps/:id/variables/`

> #### Payload

Reviewer 2: 1. table format is easy to read
2. add about base64 for binary data

To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default, the variable is encrypted. Set the value of `secure` to `false` to not encrypt the newly added variable. An example payload:

```json
{
  "key": "your variable name",
  "value": "your variable value",
  "group": "your variable group",
  "secure": true
}
```

Note that the `key` and `group` values have to be strings with a length greater than zero. Furthermore, if the group does not exist, it will be created.

#### CURL example

```bash
curl -XPOST -H 'X-Auth-Token: <API Token>' -H "Content-type: application/json" -d '{
    "key": "your variable name",
    "value": "your variable value",
    "group": "your variable group",
    "secure": true
  }' 'https://api.codemagic.io/apps/<app_id>/variables'
```

#### Response

```json
{
  "key": "your variable name",
  "value": "[HIDDEN]",
  "group": "your variable group",
  "secure": true
}
```

### Update existing variable

`POST /apps/:id/variables/:variable_id`

> #### Payload

Reviewer 2: also replace with section "Parameter" and table
